src/or/prop259.c

```diff
@@ -592,7 +592,6 @@ choose_entry_guard_algo_end(guard_selection_t *guard_selection,
 {
     if (!smartlist_contains(guard_selection->used_guards, guard))
         smartlist_add(guard_selection->used_guards, (void*) guard);
-        smartlist_add(used_guards, (void*) guard);
```

Comment:
In most cases the guard_selection will be freed, in that case guard_selection->used_guards will not be picked up after `end` happens.